labview/labview_rabbitmq.c

```diff
@@ -0,0 +1,277 @@
+	return lv_report_amqp_error(amqp_get_rpc_reply(conn), "Opening channel", error_description);
+}
+
+
+LABVIEW_PUBLIC_FUNCTION
+int lv_amqp_channel_close(int64_t conn_intptr, uint16_t  channel, LStrHandle error_description) {
+	amqp_connection_state_t conn = (amqp_connection_state_t)conn_intptr;
+	return lv_report_amqp_error(amqp_channel_close(conn, channel, AMQP_REPLY_SUCCESS), "Closing channel", error_description);
+}
+
+
+LABVIEW_PUBLIC_FUNCTION
+int lv_amqp_exchange_declare(int64_t conn_intptr, uint16_t  channel, char *exchange, char *exchangetype, LStrHandle error_description) {
+	amqp_connection_state_t conn = (amqp_connection_state_t)conn_intptr;
+
+	amqp_boolean_t PASSIVE = 0;
+	amqp_boolean_t DURABLE = 0;
+	amqp_boolean_t AUTO_DELETE = 0;
+	amqp_boolean_t INTERNAL = 0;
+	amqp_exchange_declare(conn, channel, amqp_cstring_bytes(exchange),
+		amqp_cstring_bytes(exchangetype), PASSIVE, DURABLE, AUTO_DELETE, INTERNAL,
+		amqp_empty_table);
+
+    return lv_report_amqp_error(amqp_get_rpc_reply(conn), "Exchange declare", error_description);
+}
+
+
+LABVIEW_PUBLIC_FUNCTION
+int lv_amqp_login(int64_t conn_intptr, char *host, int port, int timeout_sec, char *username, char *password, LStrHandle error_description) {
+	
+	int status;
+	amqp_socket_t *socket = NULL;
+	amqp_connection_state_t conn = (amqp_connection_state_t)conn_intptr;
+
+	struct timeval tval;
+	tval.tv_sec = timeout_sec;
+	tval.tv_usec = 0;
+
+	socket = amqp_tcp_socket_new(conn);
+	if (!socket) {
+		return _CREATING_TCP_SOCKET;
+  	}
+	status = amqp_socket_open_noblock(socket, host, port, &tval);
+	if (status<0) {
+    return _OPENING_TCP_SOCKET;
+  	} 
+	/*Code explanation:
+	socket is set/stored in the connection state,
+	it will be destroyed along with connection state destroy function*/ 
+
+	char const *VHOST = "/"; 		// the virtual host to connect to on the broker. The default on most brokers is "/"
+	int const CHANNEL_MAX = 0; 		// the limit for number of channels for the connection. 0 means no limit.
+	int const FRAME_MAX = 131072; 	// the maximum size of an AMQP frame. 131072 is the default. 
+									// 4096 is the minimum size, 2^31-1 is the maximum, a good default is 131072 (128KB),
+	int const HEARTBEAT = 0; 		// the number of seconds between heartbeat frames to request of the broker. A value of 0 disables heartbeats.
+
+	return lv_report_amqp_error(amqp_login(conn, VHOST, CHANNEL_MAX, FRAME_MAX, HEARTBEAT, AMQP_SASL_METHOD_PLAIN, username, password), "Logging in", error_description);	
+}
+
+
+LABVIEW_PUBLIC_FUNCTION
+int lv_amqp_basic_publish(int64_t conn_intptr, uint16_t  channel, char *exchange, char *routingkey, char *messagebody, LStrHandle error_description) {
+	amqp_connection_state_t conn = (amqp_connection_state_t)conn_intptr;
+	amqp_basic_properties_t props;
+	props._flags = AMQP_BASIC_CONTENT_TYPE_FLAG | AMQP_BASIC_DELIVERY_MODE_FLAG;
+	props.content_type = amqp_cstring_bytes("text/plain");
+	props.delivery_mode = 2; /* persistent delivery mode */
```

Comment:
I see it is a cop y past from original send string example. no need for us to support other option?